### PR TITLE
[master] KAZOO-5736: properly handling retention for voicemail operations

### PR DESCRIPTION
**Pull request by icehess**
4.1: #4321

### CODE REVIEW

File: core/kazoo_modb/priv/couchdb/views/mailbox_messages.json [hunk 1]
```diff
@@ -2,15 +2,70 @@
...
+            ]
+        },
         "count_per_folder": {
-            "map": "function(doc) { if (doc.pvt_type != 'mailbox_message' || doc.pvt_deleted) return; emit([doc.source_id, doc.metadata.folder], 1); }",
-            "reduce": "_count"
+            "map": [
+                "function(doc) {",
+                "  if (doc.pvt_type != 'mailbox_message' || doc.pvt_deleted)",
+                "    return;",
+                "  emit([doc.metadata.folder, doc.metadata.timestamp, doc.source_id]);",
```
Comment (jamesaimonetti) [inline core/kazoo_modb/priv/couchdb/views/mailbox_messages.json]: does `emit/1` use `null` as the value by default?

**Comment by jamesaimonetti** [comment]
@icehess i think you need to rebase and all will be well with the PR

File: core/kazoo_modb/priv/couchdb/views/mailbox_messages.json (hunk 1 above)
Comment (icehess) [reply core/kazoo_modb/priv/couchdb/views/mailbox_messages.json]: yes, though i'm not sure about bigcouch :-)

Comment (icehess) [reply core/kazoo_modb/priv/couchdb/views/mailbox_messages.json]: well I see karl added `missing_metadata` without specifying value, so it is working in bigcouch too.

**Pull request merged**